### DISCUSSION TITLE
MQTT setup with homeassistant and mosquito broker

### DISCUSSION THREAD

**Discussion by dychonn** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#668] [Q&A] [answered]
Hi, 

I made the Wifi with an Wemos D1 for my lay z S100101 pump. All is working fine and the web interface works great. Now i am starting with the MQTT connection with Home Assistant. I enabled the MQTT and set an password and username. The MQTT (Mosquitto) broker is integrated and setup with an mqtt-user. 

The web interface of the wifi controller is working fine i think, because it says connection failed or disconnected. .

I think the problem is in the home assistant settings. I am very new to this, but i have a lot of software knowledge. Maybe someone could point me a direction. The question is where do i put the settings from the mqtt device, for example the ip adress 192.168.0.20 and port 1883. That is one thing that's not clear for me, because i read here on github that it should automatically pop up in home assistant. 

Hope to hear from you guys with my new hobby. 

Dyon

**Comment by visualapproach** [comment, marked as answer]
Hi,
As you say, you should setup HA with an MQTT broker like mosquitto. Preferably also create a HA user "mymqttuser" with a password. You don't need to setup a special user in the MQTT broker. Instead go to settings/persons/users and set it up. Figure out the IP of home assistant by going to settings/system/network and check out the address under network adapter.

On the ESP side, visit the MQTT config page and enter that IP address, port 1883 unless you change it. User name and password as you created in previous step.

The things that pops up automatically is all the sensors and switches, once you establish mqtt connection. It will take a reboot of the ESP and a few minutes before it shows up though.

/Visualapproach

**Reply by dychonn** [reply]
Works very good. I can follow the steps. After 2 hours trying i get it working just before your answer. Your answer is helpfull for new people to mqtt.
